arch/aarch64/makecontext.c

```diff
@@ -32,14 +32,16 @@ _Static_assert(offsetof(libucontext_ucontext_t, uc_mcontext.pstate) == PSTATE_OF
 void
 libucontext_makecontext(libucontext_ucontext_t *ucp, void (*func)(void), int argc, ...)
 {
-	unsigned long *sp;
-	unsigned long *regp;
+	// specs require only 'int' arguments so stack arguments are int type
```

Comment:
What happens if we pass a pointer as a stack argument to makecontext then?

Reply:
Wouldn’t be allowed according to the man pages https://man7.org/linux/man-pages/man3/makecontext.3.html

> When this context is later activated (using setcontext(3) or
>       swapcontext()) the function func is called, and passed the series
>       of integer (int) arguments that follow argc; the caller must
>       specify the number of these arguments in argc.

Reply:
The NetBSD and glibc implementations also do not narrow the arguments like this code is doing.  The macOS implementation also does not appear to do so.  What problem are you trying to solve here?

Reply:
Well you need to choose some size or the stack layout won’t work. The old code doesn’t work because it assumed every register is 4 bytes so I bumped the register arguments to 8 bytes and kept the stack arguments at 4 bytes. You can make everything 8 bytes as I originally tried but then I read the man pages and realized it expects ‘int’ which is 4 bytes for all arguments.

Reply:
To clarify I didn’t “narrow” any arguments. I widened it for the register arguments. unsigned long is 4 bytes on arm and arm64.

Reply:
Yeah, I was misreading what the patch did.  This is fine.

Reply:
Wait what?

On my `aarch64` machine, `unsigned long` is definitely 8 bytes:

```
petrie:~/libucontext$ cat moo2.c
#include <stdint.h>
#include <stdio.h>

int main() {
        printf("%zu\n", sizeof(unsigned long));
        return 0;
}
petrie:~/libucontext$ gcc -o moo2 moo2.c
petrie:~/libucontext$ ./moo2
8
```

Reply:
Apple says `unsigned long` is 8 bytes on Darwin ARM64 too.  https://developer.apple.com/documentation/xcode/writing_arm64_code_for_apple_platforms

Reply:
It seems to me that having it any other way would break alignment assumptions made by aarch64 CPUs?  So I'm back to being confused about the two ARM-specific patches.